### PR TITLE
docs(agw): update network probe documentation

### PR DESCRIPTION
**Pull request by ymasmoudi**
Signed-off-by: YOUSSEF EL MASMOUDI <ymasmoudi@fb.com>

## Summary

Update network probe (LI) feature documentation

### CODE REVIEW

File: docs/readmes/howtos/network_probe.md [hunk 1]
```diff
@@ -4,11 +4,13 @@ title: Network Probe
 hide_title: true
 ---
 
-# Network Probe
+# Lawfull Interception
 
 ## Overview
 
-Network Probe allows a Magma operator to provide standardized lawful interception X2 and X3 interfaces as described in ETSI TS 103 221-2. This feature takes advantage of the rest API (swagger) to provide the X1 interface.
+Network Probe solution allows a Magma operator to provide standardized lawful interception X2 and X3 interfaces as described in ETSI TS 103 221-2. This feature takes advantage of the rest API (swagger) to provide the X1 interface.
```
Comment (Neudrino) [inline docs/readmes/howtos/network_probe.md]: `The Network Probe solution...`
Is "Network Probe" a name? Maybe
`The network probe solution...`

File: docs/readmes/howtos/network_probe.md [hunk 2]
```diff
@@ -4,11 +4,13 @@ title: Network Probe
...
+
+Standarized X1 interface is provided through an external solution.
```
Comment (Neudrino) [inline docs/readmes/howtos/network_probe.md]: Would it be possible to give a link to the external resource here?

File: docs/readmes/howtos/network_probe.md [hunk 3]
```diff
@@ -22,35 +24,43 @@ The LI feature can be summarized as follow,
 
 ### X1 Interface
 
-It relies on the Orc8r Swagger API to configure intercept tasks and destinations. This interface uses Json content and thus is not 3GPP complaint. An external solution (nprobe-proxy) can handle the translation 3GPP (XML based) <-> Orc8r Swagger When required.
+It relies on the Orc8r Swagger API to configure intercept tasks and destinations. This interface uses Json content and thus is not 3GPP complaint. An external solution (nprobe-proxy) can handle the translation 3GPP (XML based) <-> Orc8r Swagger when required.
```
Comment (Neudrino) [inline docs/readmes/howtos/network_probe.md]: - `It` is confusing for me here. We are probably better off "naming the devil by its name". E.g. `The X1 interface relies on...`
- Could we link the external solution, i.e. nprobe-proxy?
- We might want to replace `<->` by words, i.e. `between ... and ...`
- Thanks for the cleanup! (`when`) :broom:

File: docs/readmes/howtos/network_probe.md [hunk 4]
```diff
@@ -22,35 +24,43 @@ The LI feature can be summarized as follow,
...
+
+*Note that nprobe-proxy is provided outside magma project and will be described separately.*
```
Comment (Neudrino) [inline docs/readmes/howtos/network_probe.md]: This sentence instantly raises the question "Where?" with the reader. We should probably answer this question with a link to reading material.

File: docs/readmes/howtos/network_probe.md [hunk 5]
```diff
@@ -22,35 +24,43 @@ The LI feature can be summarized as follow,
...
+- target_id : represents the subscriber identifier
+- target_type : represents the subscriber identifier type (IMSI, IMEI, MSISDN). Only IMSI is supported now.
+- delivery_type : (events_only/all) states whether to deliver X2 or both X2 and X3 to the LIMS.
+- correlation_id : allows X2 and X3 records correlation. A random value is generated if not provided.
+- operator_id : operator identifier
+- domain_id : domain identifier
+- duration : specifies the lifetime of the task. If set to 0, the task will not expire until deleted through APIs.
 
-Each configured task in swagger will be then propagated to the appropriate service (nprobe, liagentd, pipelined)
+Each configured task in swagger will be propagated to the appropriate services (nprobe, liagentd, pipelined)
```
Comment (Neudrino) [inline docs/readmes/howtos/network_probe.md]: _nit-pick:_ Full stop missing for this sentence.

File: docs/readmes/howtos/network_probe.md [hunk 6]
```diff
@@ -22,35 +24,43 @@ The LI feature can be summarized as follow,
...
-- Duration: specifies the lifetime of the task. If set to 0, the task will not expire until deleted through APIs.
-- CorrelationID : allows to correlates X2 and X3 records. If not provided, Orc8r will generate a random value.
+- task_id : is UUID v4 representing an XiD identifier.
+- target_id : represents the subscriber identifier
+- target_type : represents the subscriber identifier type (IMSI, IMEI, MSISDN). Only IMSI is supported now.
+- delivery_type : (events_only/all) states whether to deliver X2 or both X2 and X3 to the LIMS.
+- correlation_id : allows X2 and X3 records correlation. A random value is generated if not provided.
+- operator_id : operator identifier
+- domain_id : domain identifier
+- duration : specifies the lifetime of the task. If set to 0, the task will not expire until deleted through APIs.
```
Comment (Neudrino) [inline docs/readmes/howtos/network_probe.md:40]: If the names of those configurations was changed shortly, could we link the respective PR in the description of this pull request please.

File: docs/readmes/howtos/network_probe.md [hunk 7]
```diff
@@ -22,35 +24,43 @@ The LI feature can be summarized as follow,
...
-- DeliveryAddress : provides the address of the remote server.
-- DeliveryType : (events_only/all) states whether the server can receive X2 or both X2 and X3.
+- delivery_address : provides the address of the remote server.
+- delivery_type : (events_only/all) states whether the server can receive X2 or both X2 and X3.
+- private_key : TLS private key to connect the delivery address
+- certificate : TLS certificate to connect to the delivery address
+- skip_verify_server : skip client verification when self-signed certificates are provided.
 
-*Note that destination configuration is not currently taken in account. Only manual config is supported.*
+*Note The orc8r nprobe service (X2 Interface) processes the first destination only. Subsequent destinations are ignored.*
```
Comment (Neudrino) [inline docs/readmes/howtos/network_probe.md]: Either `Note: The orc8r...` or `Note that the orc8r...`

Comment (Neudrino) [inline docs/readmes/howtos/network_probe.md]: _Matter of taste_: I would personally prefer to have `orchestrator` written in full instead of abbreviated `orc8r`, however, if it is consistent with the document, it is fine for me as is.

File: docs/readmes/howtos/network_probe.md [hunk 8]
```diff
@@ -22,35 +24,43 @@ The LI feature can be summarized as follow,
...
 
 ### X2 Interface
 
-It is provided by the nprobe service in Orc8r. This service collects all the relevant events for targeted subscriber through elastic search from fluentd. Then, it parses them to create X2 records (aka Intercept Related Information - IRI) as specified ETSI TS 103 221-2 before exporting them to a remote server over TLS.
+It is provided by the nprobe service in Orc8r. This service collects all the relevant events for targeted subscriber from fluentd through elastic search. Then, it parses them to create X2 records (aka Intercept Related Information - IRI) as specified ETSI TS 103 221-2 before exporting them to a remote server over TLS.
```
Comment (Neudrino) [inline docs/readmes/howtos/network_probe.md]: `It` is confusing for me here. We are probably better off "naming the devil by its name". E.g. `The X2 interface is provided...`

File: docs/readmes/howtos/network_probe.md (hunk 1 above)
Comment (Neudrino) [inline docs/readmes/howtos/network_probe.md]: We might also want to give a link to the `ETSI TS 103 221-2` regulations here.

File: docs/readmes/howtos/network_probe.md [hunk 9]
```diff
@@ -22,35 +24,43 @@ The LI feature can be summarized as follow,
...
+Current records are supported: BearerActivation, BearerModification, BearerDeactivation, EutranAttach.
```
Comment (Neudrino) [inline docs/readmes/howtos/network_probe.md]: Could we rephrase this to be a full sentence?
E.g. `Current records are supported are BearerActivation, BearerModification, BearerDeactivation and EutranAttach.`

File: docs/readmes/howtos/network_probe.md [hunk 10]
```diff
@@ -85,48 +92,26 @@ Edit /etc/magma/pipelined.yml
...
-    - li_local_iface: eth2
+    - li_local_iface: gtp_br0
     - li_mirror_all: false
     - li_dst_iface: li_port
+- restart pipelined
 
 ### 2. Enable LiAgentD service in AGW
 
-Edit /etc/magma/liagentd.yml
+Copy nprobe.{pem,key} to /var/opt/magma/certs/, then edit /etc/magma/liagentd.yml
```
Comment (Neudrino) [inline docs/readmes/howtos/network_probe.md]: We might want to use codestyle here for better readability, e.g. `nprobe.{pem,key}` and `/var/opt/magma/certs/`, etc.

File: docs/readmes/howtos/network_probe.md [hunk 11]
```diff
@@ -150,9 +136,25 @@ Go to **Swagger API**:
...
+    "skip_verify_server": false
+  },
+  "destination_id": "29f28e1c-f230-486a-a860-f5a784ab9177"
+}
+```
+
 ## Test and Troubleshooting
 
-It is recommendable that before running the tests, you enable some extra logging capabilities in both Access Gateway.
+It is recommendable that before running the tests, you enable some extra logging capabilities in both Access Gateway and Orc8r
```
Comment (Neudrino) [inline docs/readmes/howtos/network_probe.md]: _nit-pick:_ Full stop missing for this sentence.

File: docs/readmes/howtos/network_probe.md (hunk 7 above)
Comment (ymasmoudi) [reply docs/readmes/howtos/network_probe.md]: `orc8r` is not only an abbreviation of `orchestrator` but the name of a product.

File: docs/readmes/howtos/network_probe.md (hunk 6 above)
Comment (ymasmoudi) [reply docs/readmes/howtos/network_probe.md:40]: previous names reflected internal go structures attributes. 
this change correct that to use api endpoint json fields.

File: docs/readmes/howtos/network_probe.md (hunk 4 above)
Comment (ymasmoudi) [reply docs/readmes/howtos/network_probe.md]: Sure, this is still in progress.

File: docs/readmes/howtos/network_probe.md (hunk 2 above)
Comment (ymasmoudi) [reply docs/readmes/howtos/network_probe.md]: TBD once the repo is ready

File: docs/readmes/howtos/network_probe.md (hunk 6 above)
Comment (Neudrino) [reply docs/readmes/howtos/network_probe.md:40]: I see, could you still link the place / file, where they are defined in the code, please.

File: docs/readmes/howtos/network_probe.md [hunk 12]
```diff
@@ -4,11 +4,13 @@ title: Network Probe
 hide_title: true
 ---
 
-# Network Probe
+# Lawfull Interception
 
 ## Overview
 
-Network Probe allows a Magma operator to provide standardized lawful interception X2 and X3 interfaces as described in ETSI TS 103 221-2. This feature takes advantage of the rest API (swagger) to provide the X1 interface.
+The Network Probe solution allows a Magma operator to provide standardized lawful interception X2 and X3 interfaces as described in ETSI TS 103 221-2 (https://www.etsi.org/deliver/etsi_ts/103200_103299/10322102/01.04.01_60/ts_10322102v010401p.pdf). This feature takes advantage of the rest API (swagger) to provide the X1 interface.
```
Comment (Neudrino) [inline docs/readmes/howtos/network_probe.md]: Link syntax is `[Text](https://your.domain/with/link)`. I.e. brackets missing here.

File: docs/readmes/howtos/network_probe.md (hunk 2 above)
Comment (Neudrino) [reply docs/readmes/howtos/network_probe.md]: What does that mean?
And if you are just creating it, why not integrating with Magma repo?

File: docs/readmes/howtos/network_probe.md [hunk 13]
```diff
@@ -85,48 +97,26 @@ Edit /etc/magma/pipelined.yml
...
-    - li_local_iface: eth2
+    - li_local_iface: gtp_br0
     - li_mirror_all: false
     - li_dst_iface: li_port
+- restart pipelined
 
 ### 2. Enable LiAgentD service in AGW
 
-Edit /etc/magma/liagentd.yml
+Copy `nprobe.{pem,key}` to `/var/opt/magma/certs/`, then edit /etc/magma/liagentd.yml
```
Comment (Neudrino) [inline docs/readmes/howtos/network_probe.md]: Of course also the 3rd file location should have the Code style consistently.

**Comment by Neudrino** [comment]
Hi @ymasmoudi ,
any reason this was closed?
This is in general a good contribution and only needs some small styling improvements.

File: docs/readmes/howtos/network_probe.md (hunk 2 above)
Comment (ymasmoudi) [reply docs/readmes/howtos/network_probe.md]: Solved. There will be no mention to an external solution for now.

File: docs/readmes/howtos/network_probe.md (hunk 12 above)
Comment (Neudrino) [reply docs/readmes/howtos/network_probe.md]: There must not be a space character between the `]` and `(` to be syntax compliant.

File: docs/readmes/howtos/network_probe.md [hunk 14]
```diff
@@ -4,11 +4,11 @@ title: Network Probe
 hide_title: true
 ---
 
-# Network Probe
+# Lawfull Interception
 
 ## Overview
 
-Network Probe allows a Magma operator to provide standardized lawful interception X2 and X3 interfaces as described in ETSI TS 103 221-2. This feature takes advantage of the rest API (swagger) to provide the X1 interface.
+The Network Probe solution allows a Magma operator to provide standardized lawful interception X2 and X3 interfaces as described in [ETSI TS 103 221-2] (https://www.etsi.org/deliver/etsi_ts/103200_103299/10322102/01.04.01_60/ts_10322102v010401p.pdf). This feature takes advantage of the rest API (swagger) to provide the X1 interface.
```
Comment (Neudrino) [inline docs/readmes/howtos/network_probe.md]: Again: Link syntax is `[Text](https://your.domain/with/link)`.

Comment (ymasmoudi) [reply docs/readmes/howtos/network_probe.md]: fixed.

**Review by Neudrino — APPROVED** [review]
